AlphaWallet/Market/Coordinators/UniversalLinkCoordinator.swift

```diff
@@ -489,13 +490,14 @@ class UniversalLinkCoordinator: Coordinator {
 
     private func makeTokenHolderImpl(name: String, symbol: String, bytes32Tokens: [String], contractAddress: AlphaWallet.Address) {
         //TODO pass in the wallet instead
+        guard let tokenHolderStrong = tokenHolder else { return }
```

Comment:
This wouldn't work, because `self.tokenHolder` is nil. (An unfortunate part of using `TokenHolder?` as a type here). We'll need to use something like:

```
let tokensDatastore = tokensDatastores[server]
let token = tokensDatastore.token(forContract: someContract)
//token.tokenType
```

Reply:
Haha that's what I thought my happen since it is assigned below, one second.